src/modules/vtol_att_control/standard.cpp

```diff
@@ -337,78 +337,77 @@ void Standard::update_fw_state()
+		fw_out[actuator_controls_s::INDEX_AIRBRAKES]    = 0;
 
-			// pitch
-			_actuators_out_1->control[actuator_controls_s::INDEX_PITCH] =
-				_actuators_fw_in->control[actuator_controls_s::INDEX_PITCH];
+		break;
 
-			_actuators_out_1->control[actuator_controls_s::INDEX_YAW] = 0.0f;
-			_actuators_out_1->control[actuator_controls_s::INDEX_AIRBRAKES] = 0.0f;
-		}
-	}
+	case vtol_mode::TRANSITION_TO_FW:
 
-	// set the fixed wing throttle control
-	if (_vtol_schedule.flight_mode == vtol_mode::FW_MODE) {
+	// FALLTHROUGH
+	case vtol_mode::TRANSITION_TO_MC:
+		// MC out = MC in (weighted)
+		mc_out[actuator_controls_s::INDEX_ROLL]         = mc_in[actuator_controls_s::INDEX_ROLL]     * _mc_roll_weight;
+		mc_out[actuator_controls_s::INDEX_PITCH]        = mc_in[actuator_controls_s::INDEX_PITCH]    * _mc_pitch_weight;
+		mc_out[actuator_controls_s::INDEX_YAW]          = mc_in[actuator_controls_s::INDEX_YAW]      * _mc_yaw_weight;
+		mc_out[actuator_controls_s::INDEX_THROTTLE]     = mc_in[actuator_controls_s::INDEX_THROTTLE] * _mc_throttle_weight;
+		mc_out[actuator_controls_s::INDEX_LANDING_GEAR] = 0;
+
+		// FW out = FW in, with VTOL transition controlling throttle and airbrakes
+		fw_out[actuator_controls_s::INDEX_ROLL]         = fw_in[actuator_controls_s::INDEX_ROLL];
+		fw_out[actuator_controls_s::INDEX_PITCH]        = fw_in[actuator_controls_s::INDEX_PITCH];
+		fw_out[actuator_controls_s::INDEX_YAW]          = fw_in[actuator_controls_s::INDEX_YAW];
+		fw_out[actuator_controls_s::INDEX_THROTTLE]     = _pusher_throttle;
+		fw_out[actuator_controls_s::INDEX_FLAPS]        = fw_in[actuator_controls_s::INDEX_FLAPS];
+		fw_out[actuator_controls_s::INDEX_AIRBRAKES]    = _reverse_output;
 
-		// take the throttle value commanded by the fw controller
-		_actuators_out_1->control[actuator_controls_s::INDEX_THROTTLE] =
-			_actuators_fw_in->control[actuator_controls_s::INDEX_THROTTLE];
+		break;
 
-	} else {
-		// otherwise we may be ramping up the throttle during the transition to fw mode
-		_actuators_out_1->control[actuator_controls_s::INDEX_THROTTLE] = _pusher_throttle;
+	case vtol_mode::FW_MODE:
+		// MC out = 0
+		mc_out[actuator_controls_s::INDEX_ROLL]         = 0;
+		mc_out[actuator_controls_s::INDEX_PITCH]        = 0;
+		mc_out[actuator_controls_s::INDEX_YAW]          = 0;
+		mc_out[actuator_controls_s::INDEX_THROTTLE]     = 0;
+		mc_out[actuator_controls_s::INDEX_LANDING_GEAR] = 0;
+
+		// FW out = FW in
+		fw_out[actuator_controls_s::INDEX_ROLL]         = fw_in[actuator_controls_s::INDEX_ROLL];
+		fw_out[actuator_controls_s::INDEX_PITCH]        = fw_in[actuator_controls_s::INDEX_PITCH];
+		fw_out[actuator_controls_s::INDEX_YAW]          = fw_in[actuator_controls_s::INDEX_YAW];
+		fw_out[actuator_controls_s::INDEX_THROTTLE]     = fw_in[actuator_controls_s::INDEX_THROTTLE];
+		fw_out[actuator_controls_s::INDEX_FLAPS]        = fw_in[actuator_controls_s::INDEX_FLAPS];
+		fw_out[actuator_controls_s::INDEX_AIRBRAKES]    = 0;
```

Comment:
The airbrakes could also be passed through in theory in FW, no? Would actually be useful for a steep descend.

Reply:
Within VTOL standard INDEX_AIRBRAKES is used (questionably) as the control line for reversible ESCs.